src/test/resources/optimized-query-for-filter.adoc

```diff
@@ -0,0 +1,897 @@
+:toc:
+= Translator Tests
+
+== Schema
+
+[source,graphql,schema=true]
+----
+type Movie {
+  title: String!
+  released: Int
+  tagline: String
+  actor: [Person!]! @relation(name:"ACTED_IN", direction: IN)
+  reviews: [Review!]!
+  reviewedBy: [Person!]! @relation(name:"REVIEWED", direction: IN)
+  producedBy: [Person!]! @relation(name:"PRODUCED", direction: IN)
+  writtenBy: [Person!]! @relation(name:"WROTE", direction: IN)
+  directedBy: [Person!]! @relation(name:"DIRECTED", direction: IN)
+  directedBySingle: Person @relation(name:"DIRECTED", direction: IN)
+}
+type Person {
+  name: String!
+  born: Int
+  actedIn: [Movie!]! @relation(name: "ACTED_IN", direction:OUT)
+  roles: [Role!]!
+  reviewedMovies: [Movie!]! @relation(name: "REVIEWED", direction:OUT)
+  reviews: [Review!]!
+  producedMovies: [Movie!]! @relation(name: "PRODUCED", direction:OUT)
+  writtenMovies: [Movie!]! @relation(name: "WROTE", direction:OUT)
+  directedMovies: [Movie!]! @relation(name: "DIRECTED", direction:OUT)
+  follows: [Person!]! @relation(name: "FOLLOWS", direction:OUT)
+  followedBy: [Person!]! @relation(name: "FOLLOWS", direction:IN)
+}
+type Role @relation(name:"ACTED_IN", from:"actor", to:"movie") {
+  actor: Person!
+  movie: Movie!
+  roles: String
+}
+type Review @relation(name:"REVIEWED", to: "movie") {
+  from: Person!
+  movie: Movie!
+  rating: Int
+  summary: String
+}
+----
+
+== Tests
+
+=== Filter at same level (some)
+
+.Query configuration
+[source,json,config=true]
+----
+{  "optimizedQuery": true }
```

Comment:
to generic, we should use a specific term here: or an array: `optimizeQuery: ["filter-as-match"]`